src/Platforms/Gcc/UtestPlatform.cpp

```diff
@@ -76,29 +76,33 @@ static int PlatformSpecificWaitPidImplementation(int, int*, int)
 
 static void GccPlatformSpecificRunTestInASeperateProcess(UtestShell* shell, TestPlugin* plugin, TestResult* result)
 {
-    pid_t cpid, w;
+    const pid_t syscallError = -1;
+    pid_t cpid;
+    pid_t w;
     int status;
 
     cpid = PlatformSpecificFork();
 
-    if (cpid == -1) {
+    if (cpid == syscallError) {
         result->addFailure(TestFailure(shell, "Call to fork() failed"));
         return;
     }
 
     if (cpid == 0) {            /* Code executed by child */
-        shell->runOneTestInCurrentProcess(plugin, *result);   // LCOV_EXCL_LINE
-        _exit(result->getFailureCount());                     // LCOV_EXCL_LINE
+        const int initialFailureCount = result->getFailureCount(); // LCOV_EXCL_LINE
+        shell->runOneTestInCurrentProcess(plugin, *result);        // LCOV_EXCL_LINE
+        _exit(initialFailureCount < result->getFailureCount());    // LCOV_EXCL_LINE
     } else {                    /* Code executed by parent */
+        result->countRun();
         do {
             w = PlatformSpecificWaitPid(cpid, &status, WUNTRACED);
-            if (w == -1) {
-                if(EINTR ==errno) continue; /* OS X debugger */
+            if (w == syscallError) {
+                if(EINTR == errno) continue; /* OS X debugger */
                 result->addFailure(TestFailure(shell, "Call to waitpid() failed"));
                 return;
             }
 
-            if (WIFEXITED(status) && WEXITSTATUS(status) > result->getFailureCount()) {
+            if (WIFEXITED(status) && WEXITSTATUS(status) != 0) {
                 result->addFailure(TestFailure(shell, "Failed in separate process"));
             } else if (WIFSIGNALED(status)) {
                 SimpleString signal(StringFrom(WTERMSIG(status)));
```

Comment:
Do you have any idea why this signal code here (not shown in review) is so weird?

Reply:
No. But you proposed yourself to avoid touching code that I don't have to modify...

Reply:
Hah :) Right! Thanks for that :P

Reply:
as I moved it to new function, I merged few lines to make it look better (in my opinion)